server/src/main/java/org/elasticsearch/plugins/MapperPlugin.java

```diff
@@ -65,6 +66,13 @@ default Function<String, FieldPredicate> getFieldFilter() {
         return NOOP_FIELD_FILTER;
     }
 
+    /**
+     * Returns {VectorFormatProvider} implementations added by this plugin.
+     */
+    default VectorsFormatProvider getVectorsFormatProvider() {
```

Comment:
Here is my concern, do we really want this extensible by EVERY plugin or just ours? I would rather not open up the flood gates and have yet another extension that needs to be bwc and maintained until we really need it.

See `InternalSearchPlugin` as an example of this.

Reply:
Great comment,  and completely makes sense. Addressed in ae5cc15650a1d5577aaede2cfadf46b1ac2d2b2c